darknettest

SonHW opened the pull request.

TanFluent reviewed (commented):
this would solve my problem when the BB of one image more than the default 30

JaledMC committed:
Free image memory

MarvinKlemp committed:
generalized statement

MarvinKlemp committed:
Added Ampere Gencode

AlexeyAB committed:
Merge pull request #6944 from Sialitech/classifier

Restore classifier function to python API

AlexeyAB committed:
Merge pull request #6943 from MarvinKlemp/master

Added a hint for installation with CUDA version >= 11

AlexeyAB committed:
Fix nms for valid, coco 2014 -> 2017

AlexeyAB committed:
[yolo] new_coords=1

AlexeyAB committed:
added yolov4x-mish.cfg

AlexeyAB committed:
minor fix

AlexeyAB committed:
Added cspx-p7-mish.cfg

AlexeyAB committed:
max_delta=20 for yolov4x-mish.cfg

AlexeyAB committed:
fix new_coords for CPU

AlexeyAB committed:
Minor fix

AlexeyAB committed:
optimized_memory=1, learning_rate=0.001, max_delta=20, 5, 2 for yolov4x-mish.cfg

AlexeyAB committed:
Speedup [yolo] layer

AlexeyAB committed:
[net] equidistant_point=1 instead of LR-scheduler

AlexeyAB committed:
Minor fix

AlexeyAB committed:
Added yolov4-csp.cfg

AlexeyAB committed:
[net] equidistant_point=200000 num_sigmas_reject_badlabels=3 badlabels_rejection_percentage=0.1

AlexeyAB committed:
Added [net] ema_alpha=0.998

AlexeyAB committed:
Fixed Makefile and CMakeLists.txt for RTX 3090

AlexeyAB committed:
Commented in the Makefile: -gencode arch=compute_86,code=[sm_86,compute_86]

AlexeyAB committed:
Fixed new_coords=1 scale_x_y = 2.0 for yolov4x-mish.cfg and yolov4-csp.cfg

AlexeyAB committed:
[net] weights_reject_freq=1001

AlexeyAB committed:
Fix letter_box for yolov4x-mish/csp.cfg: letter_box = net.letter_box; And force for Detection: if (net.letter_box) letter_box = 1;

AlexeyAB committed:
Fixed new_coords=1 training

AlexeyAB committed:
Use default gradient for new_coords=1

AlexeyAB committed:
Fixed new_coords=1, yolov4-csp.cfg, yolov4x-mish.cfg

AlexeyAB committed:
Fixed yolov4-csp.cfg, yolov4x-mish.cfg

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Show GIoU/IoU loss instead of RMSE by default

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Save weights for each 1000 iterations if max_batches < 10000

AlexeyAB committed:
Fixed cspx-p7-mish.cfg

AlexeyAB committed:
Fixed EMA. Fixed display of GIoU / IoU loss.

AlexeyAB committed:
Reverted some changes in yolo_layer.c to fix a bug

AlexeyAB committed:
Final training fix

AlexeyAB committed:
Fix scale_xy for CPU

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Fixed equidistant point threshold

AlexeyAB committed:
Merge branch 'master' of https://github.com/AlexeyAB/darknet

AlexeyAB committed:
Minor fix for cfg-files

AlexeyAB committed:
Readme.md for Xilinx Zynq Ultrascale+ (DPU) ZCU102/ZCU104

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Update README.md

zzpbuaa committed:
Fix bug OSError: exception: access violation writing  #6897

Avoid read and write image at the same time.

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Merge pull request #7167 from zzpbuaa/zzpbuaa-patch-1

Fix bug #6897 OSError: exception: access violation writing

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Update README.md

Jmennius committed:
CMake CUDA configuration improvements (#7164)

* CMake: Fail generation if CUDA was requested but not found

..instead of resorting to CPU build silently.

This makes builds predictable and
removes the confusion of `ENABLE_CUDA` option being merely a suggestion.

Signed-off-by: Ievgen Popovych <ievgenp@seetrue.ai>

* CMake: Be more precise with ZED SDK messages

CMake generation with `ENABLE_CUDA=OFF` and `ENABLE_ZED_CAMERA=OFF`
would spit out "ZED SDK not enabled, since it requires CUDA" messages
even though ZED SDK was explicitly disabled.
Fix this by being a bit more precise with messages.

Signed-off-by: Ievgen Popovych <ievgenp@seetrue.ai>

* CMake: Honor CUDNN_ENABLE option

Setting CUDNN_ENABLE to OFF would not disable CUDNN support if
CUDNN libraries are available.

Fix this behavior by only looking for CUDNN if it was requested (CUDNN_ENABLE)
and predicate actual settings on CUDNN_FOUND.

Signed-off-by: Ievgen Popovych <ievgenp@seetrue.ai>

* github: actions: Explicitly set CUDA state in CI

cenit committed:
update GitHub actions (#7095)

* update to run-vcpkg@v6

* bump to cmake 3.18

* use newer cuda syntax, fail if cuda enabled but not found

* update win build script

* improve travis with non-cuda-enabled builds

* improve windows building process

AlexeyAB committed:
Merge pull request #4674 from nog642/fix

Minor fix

cenit committed:
default on opencv4.5 on vcpkg

cenit committed:
do not default on ninja on windows

cenit committed:
rework integrated libs logic

cenit committed:
update README

cenit committed:
add manual trigger for github actions

cenit committed:
trying to fix cuda path

cenit committed:
trying with forward slashes

cenit committed:
add cuda to path to github ci; improve diagnostic messages

cenit committed:
revert to double backslash

cenit committed:
Update vcpkg refs to OpenCV 4.5.1

cenit committed:
fix interface included directory for opencv

cenit committed:
add nasm dependency

cenit committed:
fix

cenit committed:
comment out windows-vcpkg, improve setup scripts

cenit committed:
improve manifest mode

cenit committed:
use same logic for all enables: if on, then package is required

cenit committed:
fixes

cenit committed:
do not enable cudnn feature if cuda is not enabled

cenit committed:
use same opencv feature set independently of cuda

cenit committed:
move to ubuntu 20.04 on github actions

cenit committed:
add missing pubkey install step

cenit committed:
use ninja whenever possible in build.ps1

AlexeyAB committed:
Added CUDA graphs/streams. To speed up the inference use: [net] use_cuda_graph = 1

AlexeyAB committed:
minor fix

AlexeyAB committed:
Try new docker for circle-ci

AlexeyAB committed:
Try new docker image for circle-ci

AlexeyAB committed:
Try new docker image for circle-ci

AlexeyAB committed:
circle ci fix

AlexeyAB committed:
minor fix

kk-hainq committed:
cuda: floor -> floorf, ceil -> ceilf

AlexeyAB committed:
Merge pull request #7515 from kikaitech/cuda_floorf_ceilf

Cuda: floor -> floorf, ceil -> ceilf

AlexeyAB committed:
Update README.md

AlexeyAB committed:
minor fix

AlexeyAB committed:
compile fix

AlexeyAB committed:
Update README.md

cenit committed:
Merge branch 'master' into dev/cenit/fixes

cenit committed:
upgrade to run-vcpkg@v7

cenit committed:
use build.ps1 also in CI to better test it

cenit committed:
improve non-windows handling in ps1 script

cenit committed:
find also vcpkg nested inside darknet folder

cenit committed:
use only build.ps1

cenit committed:
download weights using a local script, do not rely on vcpkg

cenit committed:
enable opencv w/cuda in build.ps1

cenit committed:
circumvent current upstream bugs

cenit committed:
improve readme wording about build.ps1

cenit committed:
improve readme more

AlexeyAB committed:
Merge pull request #7279 from cenit/dev/cenit/fixes

move to opencv4.5 on vcpkg in ci, do not default to ninja on windows, update README

cenit committed:
Merge branch 'master' into master

AlexeyAB committed:
Merge pull request #6402 from lineofbestgit/master

Small grammar change / sentence repetition fix

cenit committed:
improve build process even more (#7610)

* test for shared libs

* improve flag handling

* build so lib in ci

* clone vcpkg if not found

* fix

* improve clang handling

* bump vcpkg.json version

* gemm.c compatible with ARM

* remove unnecessary vcpkg.json field

* do not unnecessarily rebuild vcpkg in the CI build stage

* use alexeyab nuget cache

* enable self-removal of build folder

* add interactivity in build.ps1 if not opt-out

* spellcheck README

* fix another leftover spelling error

* also auto-update darknet if possible

* do not self-update darknet in CI

cenit committed:
apply feedback from vcpkg ci

renaultfernandes committed:
Fix no inference detections bug when cudnn is disabled. (#7617)

Fixes issue described here: https://github.com/AlexeyAB/darknet/issues/7616

AlexeyAB committed:
Update README.md

cenit committed:
Deploy CUDA 11.3 in CI (#7630)

* [nuget] use personal baget repo

* update cuda to 11.3 on windows

* add error messages during build to not have green badge when failed

* remove cygwin script, is untested now; use deploy-cuda.ps1 script also inside setup.ps1

cenit committed:
almost revert last PR (#7634)

* [darknet] split pr/ci pipelines to use fast caching only when accessible

* restore original pipeline names

* restore original ccpp file name

* speed up scripts avoiding -wait flag

* restore fast cache in readonly

* clean up scripts removing references to github packages, which are often down/broken

deoksangkim committed:
fix memory error in batchnorm layer (#7619)

renaultfernandes committed:
Use cublasStatus_t instead of cudaError_t when error checking cublas calls (#7626)

Cublas success is indicated by CUBLAS_STATUS_SUCCESS. This must be used
to determine if a cublas function call was successful.

Although Cuda does not complain if we use cudaError_t, when bulding with
ROCm support it is absolutely necessary to use the correct enum
(cublasStatus_t), else ROCm will always interpret the returned
status as an error and crash.

DraconisAI committed:
Fix python video (#7611)

* some bbox conversion functions

* output video in original resolution

* fixing drawing of bboxes into orig sized video

juantoca committed:
Fixed relative paths in darknet_images.py (#7566)

yildbs committed:
gaussian_yolo_layer truth size mismatch fixed due to the update of track_id (#7578)

AlexeyAB committed:
Added [empty]/[silence] and [implicit] layers

DoubleChuang committed:
fix: frame delay (#7659)

* fix: frame delay

when use low performance device such as raspberry pi or jetson nano execute darknet demo, the streaming will be delayed more and more

* fix: operate only in live stream

Co-authored-by: Double.c <double.c@inwinstack.com>

AlexeyAB committed:
Fix [implicit] layer

AlexeyAB committed:
Merge branch 'master' of https://github.com/AlexeyAB/darknet

cenit committed:
add support for Windows PowerShell (#7702)

* add support for powershell v5

* revert opencv enabled on osx

* improve error handling in initial stages

* intercept failures also when updating darknet sources themselves

* press a key before exiting script, to preserve console host window if not running in a shell

* check path length

cenit committed:
[build.ps1] improvements (#7711)

* use parentheses to make clear operator order, print out versions

* add english language pack note in README.md

* add even more build info

cenit committed:
fixes for PowerShell ISE (#7715)

cenit committed:
[build.ps1] download ninja automatically if missing + other small updates (#7721)

* self download ninja if not found on system and not requested to opt-out

* add a flag to clean up vcpkg local binary cache to force restart from scratch

* really delete if requested

cenit committed:
[build.ps1] fix 64bit build with developer powershell for VS19 (#7740)

* [build.ps1] use x64 also on developer powershell for VS

* improve handling of errors on windows platform

* [build.ps1] bump version

* improve logic handling in uselib_track

cenit committed:
[build.ps1/setup.sh] do not export vcpkg_root anymore, add setup scripts to CI (#7751)

* [build.ps1] do not export vcpkg_root anymore

* [README.md] add a build step related to Set-ExecutionPolicy on Windows

* use scope currentuser to avoid admin powershell

* simplify even more set-executionpolicy command order

* remove unnecessary decoration to shell commands to improve user experience with copy button

* write instructions to build using only cmake

* Update README.md

* add target install

* improve setup scripts for future tests

* improve setup.sh script for more tool installations and add command line flags handling

* add CI tests for setup scripts

* use nuget artifact cache to improve CI build times

* disable interactivity for build script when called from setup script, add kitware gpg key in CI

* enable cudnn in setup.sh

* add a workaround for missing nvidia drivers in CI

* setup CUDA env variables

* remove some very slow builds from travis and move them to github actions

* chmod sh scripts

* use deploy-cuda scripts also in PR pipelines

* install asm tools

* restore links missing for CI when using deploy-cuda.sh

* enable cudnn on ubuntu ci pipelines

* pass CUDA env variables

* add missing cuda install for opencv2 and opencv3 ci pipelines

* remove unnecessary steps

Co-authored-by: Alexey <AlexeyAB@users.noreply.github.com>

AlexeyAB committed:
Update README.md

cenit committed:
add missing asm tools in CI, run it every night (#7790)

* add missing asm tools

* run CI every night

* add csharp tool

Jmennius committed:
region_layer: Populate out_h/w/c fileds for consitency (#7783)

AlexeyAB committed:
fix darknet_no_gpu.vcxproj

AlexeyAB committed:
Merge branch 'master' of https://github.com/AlexeyAB/darknet

AlexeyAB committed:
Update README.md

DoubleChuang committed:
fix: flush buffer to weights-file when writing each layer (#7747)

Co-authored-by: Double.c <double.c@inwinstack.com>

cenit committed:
add automatic rebase github action (#7809)

* add automatic rebase action

* do not upload artifacts from PRs

* use proper symbol in build.ps1

* fix lib naming

* fix folder for dll artifacts

cenit committed:
[autorebase] bump to v1.5 (#7819)

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Added yolov4-p5.cfg and yolov4-p6.cfg

AlexeyAB committed:
Merge branch 'master' of https://github.com/AlexeyAB/darknet

cenit committed:
[CI] test vcpkg nightly builds (#7826)

AlexeyAB committed:
Fixed cfg: yolov4-p5.cfg, yolov4-p6.cfg. Added new Weights-files, for: yolov4-csp, yolov4-p5, yolov4-p6, yolov4x-mish

AlexeyAB committed:
Merge branch 'master' of https://github.com/AlexeyAB/darknet

AlexeyAB committed:
Fixed [<layer>] stopbackward=1, for frozen layers: don't calc batch-norm statistic and don't allocate training-buffers.

AlexeyAB committed:
Added cfg/yolov4-p5-frozen.cfg for training with pre-trained COCO-weights file yolov4-p5.weights

AlexeyAB committed:
Fixed adversarial training

AlexeyAB committed:
minor fix - added: move_window_cv() function

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Bug fix

AlexeyAB committed:
Merge branch 'master' of https://github.com/AlexeyAB/darknet

AlexeyAB committed:
bug fix

AlexeyAB committed:
Improved [net] optimized_memory=1 to reduce GPU memory usage.

AlexeyAB committed:
fix possible bug

AlexeyAB committed:
minor fix

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Update README.md

cenit committed:
fix python main module to be compiled with recent build scripts (#7876)

* fix python main module to be compiled with recent build scripts

* fixes for posix systems

AlexeyAB committed:
Added: yolov4-csp-swish.cfg, yolov4-csp-x-swish.cfg, yolov4-csp-x-swish-frozen.cfg

AlexeyAB committed:
Update README.md

yamahitsuji committed:
Fix bugs in darknet_video.py (#7908)

* fix VideoWriter frame size

* fix saved image color

AlexeyAB committed:
darknet_video.py

marcusturewicz committed:
Adds probability to precision recall curve (#7650)

7680x4320 committed:
add thresholds (#6831)

cenit committed:
[cmake] update modules (#7891)

* [cmake] update modules

* [cmake] use proper module name when using vcpkg libs

* apply fixes from review

stephanecharette committed:
allocation and error functions now take a filename, function, and line number to better diagnose where problems happen (#7925)

Nuzhny007 committed:
Add batch inference on C++ (#7915)

* Add batch inference on C++

* Return default params

* Add make_nms parameter

AlexeyAB committed:
Fixed for-loop var initialization for OpenMP, for Contrastive-loss

cenit committed:
[CI] update to cuda11.4 (#7914)

* [CI] update to cuda11.4

* update setup scripts to cuda 11.4

AlexeyAB committed:
Update README.md

sarimmehdi committed:
Made inference faster (this is especially useful when using Yolo9000) (#8009)

* Update network.c

custom_get_region_detections function now keeps track of class index with the highest probability.

* Update darknet.h

Added best_class_idx to detection struct

* Update darknet.py

added python code for faster negative removal and also faster non-max suppression

AlexeyAB committed:
fix for CUDA streams

cenit committed:
remove travis ci (#8143)

tambry committed:
Link cuda.lib using a preprocessor macro only for non-CMake (#8173)

Debian CUDA packages don't ship a shim for the Windows SO cuda.lib, so compilation fails.
For CMake it's actually linked using targets, so avoid the preprocessor macro in that case.

AlexeyAB committed:
Added yolov4-csp-s-mish.cfg, yolov4-csp-x-mish.cfg

AlexeyAB committed:
Added yolov4-csp-s-mish.cfg, yolov4-csp-x-mish.cfg

AlexeyAB committed:
Added yolov4_new.cfg

cenit committed:
update to cuda 11.5 (#8174)

cenit committed:
fix unnecessary mono call on windows (#8229)

cenit committed:
force ash shell for nuget setup

jklemmack committed:
Update YoloCSharpWrapper to support batch_size (#8249)

YoloCSharpWrapper `InitializeYolo` declaration now matches the parameters of yolo_v2_class.hpp "init".  Resolves "attempting to read/write corrupted memory" error when calling darknet.dll "detect" methods from C#.

cenit committed:
[github actions] add missing tool from base image (#8271)

cenit committed:
improve compatibility with vs2022 (#8295)

cenit committed:
[CI] build dependencies in release-only mode (#8304)

* enable release-only triplet on windows

* add debug tmate

* use release-only triplets also for linux and osx

jenhaoyang committed:
Update Readme.txt URL Typo (#8285)

Update Readme.txt Typo

stephanecharette committed:
issue #8308: memory leaks in map (#8314)

* issue #8308: memory leaks in map

* update the window title with some training stats

AdamCuellar committed:
Fix conv_lstm cuda errors and nan's (#8388)

cenit committed:
various fixes (#8398)

* re-enable opencv on osx

* enable all and all-major architecture flags for cuda

* update build script

* use proper path separator

* [windows-cuda] bump to 11.6.1

* [windows-cuda] some fixes from CI runs

* make deploy-cuda.ps1 work also when called from outside its folder

* bump to cuda 11.6 also on linux

* deploy custom cmake modules if requested by the user

* fix cmake module source path

* use proper clause for dll copy

Nuzhny007 committed:
Fixed issues 8510 and 8511 (#8516)

* Fixed issues 8510 and 8511

* Update yolo.c

Return fprintf

smartin015 committed:
Allow image_detection to receive a cv2 image; remove duplicate darknet import (#8387)

acxz committed:
- Fix warning -Werror=format-security (#8525)

Co-authored-by: Jeremy Prater <silverevo03@gmail.com>

cenit committed:
update to cuda 11.7, update pthreads4w cmake module, upgrade ps1 scripts (#8586)

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Added yolov7-tiny.cfg

AlexeyAB committed:
minor fix

AlexeyAB committed:
Update README.md

AlexeyAB committed:
Added yolov7.cfg and yolov7x.cfg

AlexeyAB committed:
fix activation silu to swish for yolov7/x.cfg

cenit committed:
restore Windows PowerShell compatibility (#8607)

* fix IsWindowsPowershell visibility

* add missing closing braces

* use proper flag

* non-interactive deploy of cuda

* set default host triplet

* add a CI test for legacy Windows PowerShell

* add powershell as shell name explicitly, otherwise github actions are already defaulting to pwsh

stephanecharette committed:
Make sure best.weights is the most recent weights for a given mAP% (#8670)

* issue #8308: memory leaks in map

* update the window title with some training stats

* make sure _best.weights is the most recent weights with that mAP%

cenit committed:
add missing dependencies (#8694)

cenit committed:
fix errors in CI (#8753)

* update build scripts

* add pkg-config on osx

* re-enable building with older opencv versions

stephanecharette committed:
better logging for filename/function/line on CUDA and CUDNN error; logging of full backtrace on Linux when an error causes Darknet to exit (#8791)

stephanecharette committed:
Update README.md

link to FAQ and Discord

stephanecharette committed:
Update README.md

Link to new repo in readme

stephanecharette committed:
Update README.md

remove bad links to google groups and reddit

cenit committed:
update build script to workaround vcpkg integrated in vs (#8818)

2b-t committed:
feat: Allow package to be used directly with ROS/ROS 2 by providing package manifest (#8783)

2b-t committed:
feat: Move set_batch_network to public API (#8782)

cenit committed:
update github actions (#8829)

* update github actions
* consolidate versions
* bump minimum cmake version

MrLIk committed:
Fix segfault for batch_detection_example (#8478)

Co-authored-by: Yuriy Savitskiy <yuriy.savitskiy@vicuesoft.com>

Commit:
Update darknet_images.py (#8559)

Safer and better type check.

imaami committed:
Update stb libraries to latest from upstream (#5941)

* Update stb_image_write to v1.14

Source-commit: 2bb4a0accd4003c1db4c24533981e01b1adfd656
Signed-off-by: Juuso Alasuutari <juuso.alasuutari@gmail.com>

* Update stb_image to v2.25

Source-commit: 0224a44a10564a214595797b4c88323f79a5f934
Signed-off-by: Juuso Alasuutari <juuso.alasuutari@gmail.com>

---------

Signed-off-by: Juuso Alasuutari <juuso.alasuutari@gmail.com>

cenit committed:
minor spelling fixes (#8830)

* minor spelling fixes
* fix spacing
* bump stb to latest and greatest

cenit committed:
add kmeansiou to build targets (#8831)

cenit committed:
upgrade to cuda 12.2 (#8832)

mhwahdan committed:
Added docker support + 1 command installation using docker-compose (#8578)

* Added cpu docker support
* add gpu docker support + docker compose for ease of use

CristiFati committed:
Fix CPU only project build failure (#8555)

CristiFati committed:
Fix thread synchronization bug + many more (#8562)

* Minor: float fps (useful for small values - e.g. CPU)
* Fix multiple code readability issues
* Better synchoronize threads (when user interrupts video)
* Better thread objects handling
* Pass data via arguments instead of global like variables - 0
* Minor code readability fixes
* CTypes definitions
* Pass data via arguments instead of global like variables - 1
* Code reordering + (minor) renames
* Pass data via arguments instead of global like variables - 2

imaami committed:
Fix format string conversion specifiers (#6192)

Signed-off-by: Juuso Alasuutari <juuso.alasuutari@gmail.com>
Co-authored-by: Stefano Sinigardi <stesinigardi@hotmail.com>

dekkmarsvin committed:
mini fix (#5662)

FileNotFoundError: Could not find module

Co-authored-by: Stefano Sinigardi <stesinigardi@hotmail.com>

mwohlm committed:
Set option ENABLE_SSE_AND_AVX_FLAGS after all dependent variables have been defined. (#8597)

CSTEZCAN committed:
Adding new cards, sorting cuda compute (#8792)

Co-authored-by: Stefano Sinigardi <stesinigardi@hotmail.com>

cenit committed:
Gl integration (#8833)

* Start adding support for running darknet from OpenGL textures

* Get CUDA-GL interop working (at least without segfaults...)

* Copy the GL texture to the CUDA input buffer

* Remove printf statements

* Code cleanup

* Fix a function call

* Remove the #ifdef GPU guard in darknet.h

* Use the width and height from the network instead of hard-coding

* Install libgles2-mesa-dev for some ubuntu targets

* Get the build working on Windows

* Add an OpenGL-CUDA integration option for CMake

* Bump the version-date in vcpkg.json

* Remove ifdef in darknet.h for the network_predict_gl_texture() function

* Use target_compile_definitions to set CUDA_OPENGL_INTEGRATION flag

* Re-add the #ifdef CUDA_OPENGL_INTEGRATION guard

---------

Co-authored-by: Brian Schwind <brianmschwind@gmail.com>

zhiliu6 committed:
Support label image filename with multiple extension. (#6667)

schedldave committed:
Configure when to write weights. (#7932)

Co-authored-by: schedldave <david.schedl@jku.at>
Co-authored-by: Schedl David <P22598@fhooe.at>
Co-authored-by: Stefano Sinigardi <stesinigardi@hotmail.com>

cenit committed:
use ansi escape sequence vt100 ESC[H ESC[J to clear screen in a portable way (win10+ supported!) (#8834)

cenit committed:
enable packaging (#8837)

mmaaz60 committed:
Add support for less or more frequent mAP calculation while training (#4918)

* Feature: Add support to pass '-mAP_epochs' from command line.

'-mAP_epochs' decide after how many epochs mAP should be calculated. Default value is 4 which means the mAP will be calculated after each 4 epochs.

jonesmwhv committed:
Fix unhandled negative resize augmentation. Propagate OpenCV errors back to C logger. (#8565)

rockcarry committed:
improve code for parser.c using int32_t for major, minor and revision of .weights file's header. (#7983)

Co-authored-by: Stefano Sinigardi <stesinigardi@hotmail.com>

umbralada committed:
added free_alphabet function; this function used for free alphabet. (#1805)

cenit committed:
cpack update (#8842)

* enable building installers in CI

* [build.ps1] improve installer handling

* do not install directly when building installer

tdurand committed:
Evo: add output to a json file option when running on a video (#3723)

cenit committed:
lower disk space usage in CI to remove spurious errors (#8843)

Co-authored-by: vsaw <hello@vsaw.net>

cenit committed:
minor code cleaning (#8844)

* minor cleanup
* vcpkg force internal tools to reduce cache misses
* move getopt to 3rdparty
* add a very small test on a couple of platform in CI

keineahnung2345 committed:
make directories of multiple layers (#3893)

cenit committed:
add missing dependencies for openvino (#8860)

* add missing automake
* add missing autoconf-archive

cenit committed:
build.ps1 is arm64 compatible (#8900)

cenit committed:
Fixes #8897 (#8901)

cenit committed:
Fixes #8885 (#8902)

K9714 committed:
Add YOLO Layer 'nbiases' (#8893)

Co-authored-by: KyeongSoo Kim <kyeongsoo.kim@konantech.com>

cenit committed:
update cuda to 12.6 (#8922)

* update cuda to 12.6
* remove cuda from opencv2 and opencv3

cenit committed:
fix runtime libs (#8924)

cenit committed:
update stb (#8926)

cenit committed:
update actions (#8930)

cenit committed:
trying to circumvent upload-artifacts@v4 breaking changes

cenit committed:
use latest ubuntu release2

cenit committed:
improve cuda handling (#8931)

* update cuda scripts to improve compatibility with latest ubuntu
* fix small regressions and update also PR script

cenit committed:
fix linking errors with opencv3

cenit committed:
add mono on osx actions runners

cenit committed:
improve win32/msc_ver compat with mingw and fix nuget https, plus add NSIS installation step (#8940)

cenit committed:
remove win-vcpkg-port ci tests, now vcpkg links to incompatible hank-ai fork (#8941)

* remove win-vcpkg-port ci tests, now vcpkg links to incompatible hank-ai fork
* remove custom build.ps1 since darknet on vcpkg is not valid anymore
* do not pre-enable cuda & other feats
